packages/hash/api/src/db/postgres/types.ts

```diff
@@ -3,6 +3,54 @@ import {
   DatabaseTransactionConnectionType,
 } from "slonik";
 
+// The variant kinds are distinct types.
+// Enum and union types unify at the top level type, which isn't desired for these typings.
+export type PoolConnectionKind = "PoolConnectionKind";
+export type TransactionConnectionKind = "TransactionConnection";
```

Comment:
I am not too happy with having to do this. Unfortunately both union types and enums would behave weirdly in the tagged sum type, which would try to unify types in a strange way. If anyone has a good way to allow runtime tags, please do tell!